core/src/main/scala/akka/persistence/dynamodb/internal/BySliceQuery.scala

```diff
@@ -71,11 +71,8 @@ import org.slf4j.Logger
       else latest
 
     def nextQueryFromTimestamp(backtrackingWindow: JDuration): Instant =
-      if (backtracking) {
-        if (latest.timestamp.minus(backtrackingWindow).isAfter(latestBacktracking.timestamp))
-          latest.timestamp.minus(backtrackingWindow)
```

Comment:
So the hypothesis is that a normal query moved the offset forward more than the backtracking window:
- latestBacktracking received events until 12:08:07
- normal query receives events until 12:10:20
- then new backtracking query from 12:08:20
  - meaning that there is a gap of 3 seconds in the backtracking

However, that could happen after a restart too. After a restart the backtracking only goes back the backtracking window (plus backtrackingBehindCurrentTime). The assumption is that the backtracking window should cover all late arrivals.

I don't know if we could see if parts of the GSI was more far behind than the backtracking window.

Would it make sense to instead increase the backtracking window?

Reply:
Would be useful if we can confirm the GSI was actually minutes behind. Maybe the AWS team have internal visibility into this.

Another scenario is around the backtracking behind-current-time (10 seconds by default), which actually sets how late events can be. If parts of the GSI are over 10 seconds behind on updates, the backtracking queries can move ahead of this, and events will be missed (rather than behind the backtracking window).

Reply:
Yes, I see the same. It would be any kind of partial updates to the GSI within a slice that could create problems. I'd expect that changes could be at the level of DynamoDB's internal partitions, and throttling is applied within those partitions, so seems possible that there could be partition-level delays that would be partial for a slice.